service/history/queue/timer_queue_processor.go

```diff
@@ -519,6 +519,18 @@ func newTimerQueueStandbyProcessor(
 		if !ok {
 			return false, errUnexpectedQueueTask
 		}
+		if timer.TaskType == persistence.TaskTypeWorkflowTimeout ||
+			timer.TaskType == persistence.TaskTypeDeleteHistoryEvent {
+			domainEntry, err := shard.GetDomainCache().GetDomainByID(timer.DomainID)
+			if err == nil {
+				for _, cluster := range domainEntry.GetReplicationConfig().Clusters {
+					if cluster.ClusterName == clusterName {
+						// guarantee the processing of workflow execution history deletion
+						return true, nil
```

Comment:
Returning true here bypasses the VerifyStandbyTask call below. We should check for the negative case only here and let the next step decide if the task is good to go.

Reply:
Bypassing verifyStandbyTask is by design. Previously we always execute these 2 type of tasks even if verifyStandbyTask is false. The improvement is to not execute the tasks if the domain is not replicated to the cluster. For example, previously a task from local domain is executed in standby queue, but with this change it won't.

Reply:
What I mean is that returning true here is a definitive decision to execute the task; although it may be invalid according to the logic in VerifyStandbyTask. 

I think what we should do is to check the timer.TaskType only and return **false** as needed (regardless of the replication config). Then do the replication config check in the VerifyStandbyTask, since it applies to both timer and transfer tasks. Finally, return true in only one place, which is at the end of the VerifyStandbyTask.

Reply:
Chatted offline and decided to keep this as-is for now.